[Feat] 콘서트 상세 화면 콘서트 정보 탭 구현

youz2me opened the pull request:
<!--
Prefix [#이슈번호] 작업 설명
예시 : Feat [#33] 마이페이지 뷰 구현
-->

# *PULL REQUEST*

## 📄 작업 내용
<!-- 작업한 내용을 두괄식으로 작성해주세요 -->
- 콘서트 상세 화면의 콘서트 정보 탭 UI를 구현했어요.
- 티켓 사이트에서 복귀 시 표시되는 플로팅 배너를 추가했어요.
- ThumbnailCard를 DSKit 공통 컴포넌트로 분리했어요.

|    구현 내용    |   정보 제보 눌렀을 때   |   티켓 사이트 복귀 배너   |   콘서트 정보 탭 오버뷰   |
| :-------------: | :----------: | :----------: | :----------: |
| GIF | <img src = "https://github.com/user-attachments/assets/ffb7059b-3312-4827-81c5-4c34cc32b4ff"> | <img src = "https://github.com/user-attachments/assets/307dc65f-629e-4d00-ad2e-132e914d877d"> | <img src = "https://github.com/user-attachments/assets/9838566e-d89f-47ee-b8dd-af298c29c09c"> |

## 💻 주요 코드 설명
<!-- 코드 설명, 없다면 생략해도 됩니다! -->
### 티켓 사이트 복귀 배너 구현
- Safari에서 티켓 사이트 방문 후 돌아오면 5초간 플로팅 배너를 표시해야 했습니다.
- 아래와 같이 구현했으며, 스와이프로 배너를 없앨 수 있습니다.

```swift
struct TicketReturnBanner: View {
    @State private var offset: CGFloat = 0
    private let autoDismissDelay: TimeInterval = 5.0
    private let dismissThreshold: CGFloat = 50

    // 아래로 50pt 이상 스와이프하면 닫힘
    .gesture(
        DragGesture()
            .onChanged { value in
                offset = max(0, value.translation.height)
            }
            .onEnded { value in
                if value.translation.height > dismissThreshold {
                    onDismiss()
                }
            }
    )
}
```

- Coordinator에서 Safari dismiss 시 콜백을 트리거하도록 구현했습니다.

```swift
case .ticketSafari(let url):
    let safariView = SafariView(url: url) { [weak self] in
        self?.dismiss()
        self?.onTicketSiteReturn?()  // 복귀 배너 표시
    }
```

### ThumbnailCard 컴포넌트 구현
- MD와 셋리스트 모두에서 사용할 수 있는 범용 카드 컴포넌트를 구현했습니다.
- `flexible` 옵션으로 고정/유동 너비를 지원합니다.

```swift
public struct ThumbnailCard: View {
    private let flexible: Bool  // true: 그리드용, false: 가로 스크롤용

    public var body: some View {
        VStack(alignment: .leading, spacing: 8) {
            thumbnailImage
            Text(title)
            if let subtitle, !subtitle.isEmpty {
                Text(subtitle)
            }
        }
        .frame(width: flexible ? nil : 108, alignment: .top)
        .frame(maxWidth: flexible ? .infinity : nil)
    }
}
```

- flexible 모드에서 이미지가 컨테이너를 벗어나지 않도록 Color.clear + overlay 기법을 사용합니다.

```swift
// 이미지 크기 제한을 위한 컨테이너 기법
Color.clear
    .aspectRatio(108/158, contentMode: .fit)
    .overlay {
        AsyncImageView(url: imageURL)
            .clipShape(RoundedRectangle(cornerRadius: 6))
    }
    .clipped()
```

## 🔗 연결된 이슈
- Resolved: LIVD-56

youz2me committed:
[Refactor] #94 LIVD-56 - SectionHeaderView 제네릭 개선 (TrailingContent, badgeSuffix)

youz2me committed:
[Refactor] #94 LIVD-56 - AsyncImageView 제네릭 Placeholder 지원

youz2me committed:
[Feat] #94 LIVD-56 - ConcertInfo Entity 추가

youz2me committed:
[Feat] #94 LIVD-56 - ScheduleRowView 컴포넌트 추가 (D-day 칩, 날짜 포맷팅)

youz2me committed:
[Feat] #94 LIVD-56 - ConcertInfoTabView 구현 (일정, 티켓 웹사이트, 콘서트 정보 캐러셀, MD 섹션)

youz2me committed:
[Fix] #94 LIVD-56 - ConcertInfoCarousel 개선 (순환 스와이프, 중복 id 수정, MD 패딩)

youz2me committed:
[Feat] #94 LIVD-56 - 일정 셀 시간 없을 때 날짜만 표시

youz2me committed:
[Feat] #94 LIVD-56 - 콘서트 상세 탭 API 연결 (일정, 필수 정보, MD)

youz2me committed:
[Fix] #94 LIVD-56 - ISO8601 DateFormatter 타임존 수정 (UTC → KST)

youz2me committed:
[Feat] #94 LIVD-56 - ThumbnailCard DSKit 컴포넌트 추가

youz2me committed:
[Feat] #94 LIVD-56 - 티켓 사이트 복귀 배너 및 카드 클릭 기능 추가

youz2me committed:
[Feat] #94 LIVD-56 - ConcertView 빈 화면 및 티켓 복귀 배너 오버레이 추가

youz2me committed:
[Feat] #94 LIVD-56 - MD 상세 화면 구현 및 ThumbnailCard flexible 옵션 추가

youz2me committed:
[Fix] #94 LIVD-56 - ThumbnailCard 레이아웃 수정 및 MD 상세 그리드 상단 정렬

JinUng41 reviewed (approved):
복잡한 UI라 그런가 코드가 많군요.
고생많으십니다 화이팅~

youz2me committed:
[Refactor] #94 LIVD-56 - ConcertInfoCarousel 분리 및 레이아웃 개선

- ConcertInfoCarousel을 별도 파일로 분리
- 탭/스와이프 제스처 분리 (onTapGesture + simultaneousGesture)
- LazyVStack 내 이미지 오버플로우 수정 (Color.clear 컨테이너 방식)
- merchandiseSection 패딩 구조 정리

youz2me committed:
[Refactor] #94 LIVD-56 - TicketReturnBanner Task 취소 로직 개선

youz2me committed:
[Refactor] #94 LIVD-56 - 포스터 오버레이 및 coordinator 콜백 정리

youz2me committed:
[Style] #94 LIVD-56 - 세그먼트 탭바 높이 조정
